Tools/debug/CollectSystemInfo.ps1

```diff
@@ -0,0 +1,248 @@
+            } catch {
+                Write-Warning "Failed to collect ${path}\${value}: $_"
+            }
+        }
+    }
+}
+
+function Get-WindowsUpdateLogs {
+    try {
+        $logPath = Join-Path -Path (Get-Location) -ChildPath "WindowsUpdate.log"
+        if (Get-Command -Name Get-WindowsUpdateLog -ErrorAction SilentlyContinue) {
+            $scriptBlock = {
+                param($logPath)
+                Get-WindowsUpdateLog -LogPath $logPath
+            }
+            Start-Process -FilePath powershell.exe -ArgumentList "-NoLogo", "-NoProfile", "-Command", "& { & $scriptBlock -logPath '$logPath' }" -NoNewWindow -Wait -RedirectStandardOutput $null -RedirectStandardError $null
+        } else {
+            Write-Warning "Get-WindowsUpdateLog cmdlet not available. Unable to collect Windows Update logs."
+        }
+    } catch {
+        Write-Warning "Failed to collect Windows Update logs: $_"
+    }
+}
+
+function Get-WindowsUptime {
+    try {
+        $uptime = (Get-Date) - (gcim Win32_OperatingSystem).LastBootUpTime
+        $uptime.ToString() | Out-File -FilePath WindowsUptime.txt
+        Write-Host "Windows uptime collection completed."
+    } catch {
+        Write-Warning "Failed to collect Windows uptime: $_"
+    }
+}
+
+function Get-ServicesList {
+    try {
+        Get-Service | Select-Object -Property Name, DisplayName, Status, StartType | Export-Csv -Path Services.csv -NoTypeInformation
+    }  catch {
+        Write-Warning "Failed to collect list of services: $_"
+    }
+}
+
+function Get-RunningProcesses {
+    try {
+        Get-Process | Select-Object -Property Id, ProcessName, StartTime | Export-Csv -Path RunningProcesses.csv -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of running processes: $_"
+    }
+}
+
+function Get-InstalledApplications {
+    try {
+        Get-ItemProperty -Path 'HKLM:\Software\Microsoft\Windows\CurrentVersion\Uninstall\*' |
+        Select-Object -Property DisplayName, DisplayVersion, Publisher, InstallDate |
+        Export-Csv -Path InstalledApplications.csv -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of installed applications: $_"
+    }
+}
+
+function Get-InstalledKBs {
+    try {
+        Get-HotFix | Select-Object -Property Description, HotFixID, InstalledOn | Export-Csv -Path InstalledKBs.csv -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of installed KBs: $_"
+    }
+}
+
+function Get-WindowsMemoryDump {
+    if (Test-Path -Path "$env:SystemRoot\MEMORY.DMP") {
+        try {
+            $zipFile = ".\MEMORY.zip"
+            Compress-Archive -Path "$env:SystemRoot\MEMORY.DMP" -DestinationPath $zipFile -Force
+        } catch {
+            Write-Warning "Failed to collect memory dump file: $_"
+        }
+    } else {
+        Write-Host "No MEMORY.DMP file found."
+    }
+    
+    $minidumpPath = "$env:SystemRoot\Minidump"
+    if (Test-Path -Path $minidumpPath) {
```

Comment:
It is a bad pattern to do `Test-Path` before performing an operation because files may become gone after `Test-Path` and before the operation. While it is unlikely to happen in this particular case, it's better to avoid the pattern whenever possible. You can catch a specific exception instead.